packages/dd-trace/src/telemetry/dependencies.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const path = require('path')
+const parse = require('module-details-from-path')
+const requirePackageJson = require('../require-package-json')
+const { sendData } = require('./send-data')
+const dc = require('diagnostics_channel')
+
+const savedDependencies = []
+const detectedDependencyNames = new Set()
+const FILE_PATH_START = `file:${path.sep}${path.sep}`
+const moduleLoadStartChannel = dc.channel('dd-trace:moduleLoadStart')
+
+let timeout, config, application, host
+
+function waitAndSend (config, application, host) {
```

Comment:
Why wait based on time? This could be too long or too small and incomplete. Can we send updates instead with new modules when they are loaded?

Reply:
This `waitAndSend` is only to avoid sending multiple messages and send only one. If the customer loads 10 libraries on startup, we can send only one message instead of 10 if we wait a bit.

Reply:
Why the delay though? Startup requires are all synchronous so a simple `setImmediate` would do, and further requires should be infrequent and sparse. This is especially true since the current logic calls the function again every 1000 entries, so in cases with thousands of dependencies it would take a very long time to finish.

Reply:
changed to setImmediate